### PR TITLE
Add deprecation warning for invalid git+http URIs

### LINKED ISSUE #8881: npm@2.12.1 onward - git+https is broken
Hey there,

it seems that the changes
e7f56e5a97fcf1c52d5c5bee71303b0126129815 and the potential fix 94ca4a711619ba8e40ce3d20bc42b13cdb7611b7
do not fix the issue with git+https

npm@2.12.0 works for me while npm@2.12.1 doesn't

It's failing with fatal: ``Unable to find remote helper for 'git+https'``

npm-debug.log is here: http://codeshare.io/UV75C

dependecy is used like this
```js
"dependencies": {
    "normalize.less": "^1.0.0",
    "patternplate": "git+ssh://git@git.sinnerschrader.com/patternplate/patternplate.git#v0.11.7",
    "react": "^0.13.3"
  }
```
Thanks for looking into this, if any information is missing, I'll try to provide it! :)
Cheers,
Tim

### PR DESCRIPTION
**Pull request by zkat**
Fixes #8881 

Main changes were made in npm/normalize-git-url#3 and npm/normalize-git-url#4

### CODE REVIEW

**Comment by othiym23** [comment]
A test, please? :smile_cat: Otherwise, :sheep::package:.

**Comment by zkat** [comment]
I need to get npm/normalize-git-url#3 merged in before this passes, but I may as well add the test case now :)

File: test/tap/add-remote-git-get-resolved.js [hunk 1]
```diff
@@ -54,16 +61,16 @@ test('add-remote-git#get-resolved SSH', function (t) {
 })
 
 test('add-remote-git#get-resolved HTTPS', function (t) {
-  var getResolved = require('../../lib/cache/add-remote-git.js').getResolved
-
   verify('https://github.com/foo/repo')
   verify('https://github.com/foo/repo#master')
   verify('git+https://github.com/foo/repo.git#master')
   verify('git+https://github.com/foo/repo#decadacefadabade')
+  // DEPRECATED
```
Comment (othiym23) [inline test/tap/add-remote-git-get-resolved.js]: `// DEPRECATED: git doesn't know what to do with https://github.com:foo/repo.git` – future us is going to be confused if we don't include at least a brief explanation, now that we've removed the deprecation warning.

**Comment by othiym23** [comment]
One suggested tweak to a comment, otherwise :rocket:.

**Comment by zkat** [comment]
better? :D

**Comment by othiym23** [comment]
Yep! :megaman: :package:

**Comment by zkat** [comment]
Merged as edbae68

**Pull request closed without merging**